### PR TITLE
feat: add forward/rewind controls for hls streaming

### PR DESCRIPTION
**Pull request by amar-1995**

### CODE REVIEW

File: packages/roomkit-react/src/Prebuilt/components/HMSVideo/PlayButton.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import React, { MouseEvent } from 'react';
+import { PauseIcon, PlayIcon } from '@100mslive/react-icons';
+import { IconButton, Tooltip } from '../../..';
+import { useHMSPlayerContext } from './PlayerContext';
+
+export const PlayButton = ({
```
Comment (raviteja83) [inline packages/roomkit-react/src/Prebuilt/components/HMSVideo/PlayButton.tsx:6]: should be named PlayPauseButton

File: packages/roomkit-react/src/Prebuilt/components/ConferenceScreen.tsx [hunk 2]
```diff
@@ -142,11 +141,6 @@ export const ConferenceScreen = () => {
             <Header />
           </Box>
         )}
-        {isMwebHLSStream && (
```
Comment (raviteja83) [inline packages/roomkit-react/src/Prebuilt/components/ConferenceScreen.tsx:145]: why is this removed?

File: packages/roomkit-react/src/Prebuilt/components/HMSVideo/HMSVideo.jsx [hunk 3]
```diff
@@ -36,6 +40,8 @@ export const HMSVideo = forwardRef(({ children, ...props }, videoRef) => {
           minHeight: '0',
           objectFit: 'contain',
           width: 'inherit',
+          height: isLandscape || isMobile ? '100%' : '',
```
Comment (raviteja83) [inline packages/roomkit-react/src/Prebuilt/components/HMSVideo/HMSVideo.jsx:43]: use of hooks seems unnecessary when this can be done with media queries

File: packages/roomkit-react/src/Prebuilt/components/ConferenceScreen.tsx (hunk 2 above)
Comment (amar-1995) [reply packages/roomkit-react/src/Prebuilt/components/ConferenceScreen.tsx:145]: we added this in hlsView itself

File: packages/roomkit-react/src/Prebuilt/components/HMSVideo/HMSVideo.jsx (hunk 3 above)
Comment (amar-1995) [reply packages/roomkit-react/src/Prebuilt/components/HMSVideo/HMSVideo.jsx:43]: it can't be done.. We are checking for media query as well as if it is mobile device or not

Comment (raviteja83) [reply packages/roomkit-react/src/Prebuilt/components/HMSVideo/HMSVideo.jsx:43]: as discussed for the previous PR, this should be done @amar-1995

File: packages/roomkit-react/src/Prebuilt/layouts/SidePane.tsx [hunk 4]
```diff
@@ -114,7 +114,7 @@ const SidePane = ({
             p: VB ? '$10 $6 $10 $10' : '$10',
             flex: '1 1 0',
             minHeight: 0,
-            maxHeight: mwebStreamingChat || isLandscape ? '300px' : 'unset',
+            maxHeight: mwebStreamingChat || isLandscape ? '340px' : 'unset',
```
Comment (raviteja83) [inline packages/roomkit-react/src/Prebuilt/layouts/SidePane.tsx:117]: maxWidth is 340 is not maxHeight. I think this is for overlay chat

Comment (raviteja83) [reply packages/roomkit-react/src/Prebuilt/layouts/SidePane.tsx:117]: should be modified on the parent width w: $100

File: packages/roomkit-react/src/Prebuilt/layouts/HLSView.jsx [hunk 5]
```diff
@@ -374,49 +377,56 @@ const HLSView = () => {
...
+                          align="center"
+                          justify="center"
                           css={{
-                            position: 'absolute',
-                            top: '40%',
-                            left: '50%',
-                            transform: 'translateY(-40%) translateX(-50%)',
-                            padding: '$4',
+                            bg: '#00000066',
+                            // transform: 'translateY(-40%) translateX(-50%)',
```
Comment (raviteja83) [inline packages/roomkit-react/src/Prebuilt/layouts/HLSView.jsx:394]: delete if not needed

File: packages/roomkit-react/src/Prebuilt/layouts/HLSView.jsx [hunk 6]
```diff
@@ -374,49 +377,56 @@ const HLSView = () => {
...
-                            >
-                              <PauseIcon width="48px" height="48px" />
-                            </IconButton>
-                          )}
-                        </Box>
+                          <HMSVideoPlayer.Seeker
+                            title="backward"
+                            css={{
+                              visibility: isSeekEnabled ? `` : `hidden`,
+                              opacity: isSeekEnabled ? `1` : '0',
```
Comment (raviteja83) [inline packages/roomkit-react/src/Prebuilt/layouts/HLSView.jsx:407]: should be visible when double clicked if already visibile. if hidden it willl be hidden.

File: packages/roomkit-react/src/Prebuilt/layouts/HLSView.jsx [hunk 7]
```diff
@@ -271,7 +273,7 @@ const HLSView = () => {
     hmsActions.setAppData(APP_DATA.hlsStats, !enablHlsStats);
   };
 
-  useEffect(() => {
+  /* useEffect(() => {
```
Comment (raviteja83) [inline packages/roomkit-react/src/Prebuilt/layouts/HLSView.jsx:276]: delete if not needed

File: packages/roomkit-react/src/Prebuilt/layouts/HLSView.jsx [hunk 8]
```diff
@@ -288,32 +290,33 @@ const HLSView = () => {
...
+      clearTimeout(controlsTimerRef.current);
+    }
+  }, []);
   const onHoverHandler = useCallback(
     event => {
+      event.preventDefault();
+      if (isMobile || isLandscape) {
+        return;
+      }
       // normal scemnario
```
Comment (raviteja83) [inline packages/roomkit-react/src/Prebuilt/layouts/HLSView.jsx:320]: what is normal scenario? typo as well

File: packages/roomkit-react/src/Prebuilt/layouts/HLSView.jsx [hunk 9]
```diff
@@ -477,8 +492,8 @@ const HLSView = () => {
                       opacity: controlsVisible ? `1` : '0',
                     }}
                   >
-                    {!(isMobile || isLandscape) && hlsState?.variants[0]?.playlist_type === HLSPlaylistType.DVR && (
-                      <HMSVideoPlayer.Progress />
+                    {!isMobile && (
```
Comment (raviteja83) [inline packages/roomkit-react/src/Prebuilt/layouts/HLSView.jsx:495]: we could have done flex-direction: column-reverse without the isMobile checks

**Review by raviteja83 — APPROVED** [review]
